### PR TITLE
TCOTA-514 TCOTA-547 enable PUT endpoint for treatments

### PR DESCRIPTION
**Pull request by relong76**
Added the service layer behind the PUT treatment endpoint to enable that service for the UI. Added new unit tests as well.

### CODE REVIEW

File: src/main/java/edu/iu/terracotta/controller/app/TreatmentController.java [hunk 1]
```diff
@@ -140,13 +137,12 @@ public ResponseEntity<Void> updateTreatment(@PathVariable("experiment_id") Long
...
 
-        if(apijwtService.isInstructorOrHigher(securedInfo)) {
-            return new ResponseEntity("Currently there is no functionality at this endpoint.", HttpStatus.OK);
-        }else {
+        if(!apijwtService.isInstructorOrHigher(securedInfo)) {
             return new ResponseEntity(TextConstants.NOT_ENOUGH_PERMISSIONS, HttpStatus.UNAUTHORIZED);
         }
-    }
 
+        return new ResponseEntity("Currently there is no functionality at this endpoint.", HttpStatus.OK);
```
Comment (machristie) [inline src/main/java/edu/iu/terracotta/controller/app/TreatmentController.java:144]: Should this call the TreatmentService.putTreatment method here?

**Review by machristie — CHANGES_REQUESTED** [review]
Thanks @relong76 . This all looks good, but see my comment. I think maybe the controller method is missing a call to the service.

File: src/main/java/edu/iu/terracotta/controller/app/TreatmentController.java (hunk 1 above)
Comment (relong76) [reply src/main/java/edu/iu/terracotta/controller/app/TreatmentController.java:144]: LOL! I'll blame the lack of coffee on this one! :)

**Review by machristie — APPROVED** [review]
👍

**Pull request merged**